src/server/saved_objects/client/lib/search_dsl/query_params.js

```diff
@@ -45,8 +45,9 @@ export function getQueryParams(mappings, type, search, searchFields) {
   if (search) {
     bool.must = [
       {
-        simple_query_string: {
+        query_string: {
```

Comment:
Maybe explain the reason for this a bit? @elastic/kibana-discovery any idea how we can prevent this from being necessary?

Reply:
Would need to know requirements

Reply:
I changed this so we can support letting the user prefix the query with a wildcard, ([request](https://github.com/elastic/kibana/issues/9767)).

So, if we had the following four visualizations stored: `My Vis`, `My_Vis`, `My-Vis`, `MyVis`, the user could query against `*Vis` and return all four results.

If we use `simple_query_string`, it _seems to_ filter out the wildcard and only returns two results (`My Vis` and `My-Vis`).

This mirrors the search behavior you'd see in a text editor which felt natural and right to me, but it's fair to suggest we do not want to support it.

Reply:
Yeah, leading wildcard is generally regarded as a problem. I think that for Kibana's use case it would make sense to use a tokenizer that converts `My Vis`, `My_Vis`, `My-Vis`, and `MyVis` all into `my` and `vis`... Is that an option given the fact that we're going to reindex objects anyway?

Reply:
Ngrams could be helpful https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-ngram-tokenizer.html

But I also wonder whether a leading wildcard will really be a problem over the set of vis documents. .kibana is generally going to be really small. We could add a config to turn wildcards off if we're really paranoid.

Reply:
That's interesting @Bargs. I'll play with that.

Reply:
I'm not concerned about the performance of leading wildcards in saved objects, but I am concerned about switching away from simple_query_string, didn't know you could configure simple_query_string to allow leading wildcards

Reply:
I don't think we can support leading wildcards with just `simple_query_string`:
`"reason": "[simple_query_string] unsupported field [allow_leading_wildcard]"`

@spalger Can you help me understand the dangers of switching from `simple_query_string` to `query_string`?

The docs say:
> Unlike the regular query_string query, the simple_query_string query will never throw an exception, and discards invalid parts of the query.

But it feels we easily mitigate the exception by catching it within Kibana server and returning an empty result set?

Reply:
I don't fully understand the ramifications of moving from `simple_query_string` to `query_string`, but I'm almost certain it's more than just about ignoring errors... Maybe @dakrone could help us understand the impact switching might have on Kibana?

Reply:
Coming into this with no history, so I'm not really sure exactly where this is going to be used but...

- SQS allows you to turn features on and off using `flags` (not sure if this is useful to you)
- SQS tries its hardest not to throw exceptions when using invalid query syntax, QS does no such thing and will flip out at the slightest syntax error (again, if you're programatically generating things it might be fine)
- SQS doesn't support every feature that QS does (like the leading wildcard, as you've been discussing)
- SQS and QS have different syntax, something like `(foo | bar) + baz` being different than `(foo OR bar) AND baz`

To weigh in my unsolicited opinion, If you are comfortable with ngrams as @bargs recommended, that would be a much easier way to support sub-string search without resorting to fancy querying. You could even have multiple fields and search both, just boosting the non-ngram version.

Hopefully that helps, happy to clarify more if I can.

Reply:
Thanks for the information @dakrone!! 

I'll continue to investigate ngrams, but it feels like, for this use case, switching from SQS to QS is a viable option still.